### PR TITLE
Cli Display

### PR DESCRIPTION
**Pull request by DaisyMolving**
Testing inputs and outputs.

Would love feedback to see if there are different, or better, ways to do this!

### CODE REVIEW

File: src/test/java/CliDisplayTest.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+    @Before
+    public void beginIOStreams() {
+        System.setOut(new PrintStream(output));
+        System.setIn(input);
+    }
+
+    @After
+    public void wipeStreams() {
+        System.setOut(null);
```
Comment (mateuadsuara) [inline src/test/java/CliDisplayTest.java]: Any `System.out.print` calls from here might blow up by `NullPointerException` or might not print anything to standard output...

Not sure if JUnit handles that well and will not be a problem.

If it is a problem, `System.out` will give you the current output stream. That way you can save it in the `@Before` and restore it here.

File: src/test/java/CliDisplayTest.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    public void cliDisplayPrintsOutputToConsole() {
+        CliDisplay cliOutput = new CliDisplay();
+        cliOutput.printMessage("Hi There!");
+        assertEquals("Hi There!", output.toString());
+    }
+
+    @Test
+    public void cliDisplayGetsInputFromConsole() {
+        CliDisplay cliInput = new CliDisplay();
+        assertEquals("string data", cliInput.getResponse("Hi There!"));
```
Comment (mateuadsuara) [inline src/test/java/CliDisplayTest.java]: Is the `"Hi There!"` relevant when a call to `getResponse` happens?

If so, what about introducing another assertion (in this test or another one) that shows where the `"Hi There!"` ends?

If not, what about removing that argument?

File: src/main/java/CliDisplay.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+    public void sendMessageToDisplay(String message) {
+       System.out.print(message);
+    }
+
+    public String getResponse(String request) {
+        Scanner userInput = new Scanner(System.in);
+        System.out.print(request);
+        return userInput.nextLine();
+    }
```
Comment (fabientownsend) [inline src/main/java/CliDisplay.java:13]: This method has two responsibilities. It display a request and return the user input

File: src/main/java/Display.java [hunk 4]
```diff
@@ -0,0 +1,5 @@
+public interface Display {
+
+    public void sendMessageToDisplay(String message);
```
Comment (fabientownsend) [inline src/main/java/Display.java:3]: You have two verbs in this method name which make me wonder if it couldn't be improve. Like for example: displayMessage, but the interface name is already *Display* so may be something like: printMessage.

As I'm still learning about naming stuff, I'm curious about your opinion about it! :)

File: src/test/java/CliDisplayTest.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+        systemOutput = System.out;
+        System.setOut(new PrintStream(output));
+        System.setIn(input);
+    }
+
+    @After
+    public void wipeStreams() {
+        System.setOut(systemOutput);
+        System.setIn(System.in);
+    }
```
Comment (fabientownsend) [inline src/test/java/CliDisplayTest.java:26]: Your input method only belong to your beginIOStreams and systemOutput to wipeStreams.

**Review by fabientownsend — COMMENTED** [review]
👍

File: src/main/java/CliDisplay.java [hunk 6]
```diff
@@ -0,0 +1,15 @@
+import java.util.Scanner;
+
+public class CliDisplay implements Display {
+
+    public void sendMessageToDisplay(String message) {
+       System.out.print(message);
```
Comment (sarahabimay) [inline src/main/java/CliDisplay.java:6]: You have System.out.print here and in the method below. If you have this method then why not use it below?

File: src/main/java/CliDisplay.java (hunk 3 above)
Comment (sarahabimay) [reply src/main/java/CliDisplay.java:13]: I could agree with this, but splitting it out will mean the calling class will have to make two method calls to CLIDisplay which can be a bit messy.  Its a tradeoff! 
eg
sendMessage("Next move please:")
nextMove = getResponse()

File: src/test/java/CliDisplayTest.java [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+    @Before
+    public void beginIOStreams() {
+        systemOutput = System.out;
+        System.setOut(new PrintStream(output));
+        System.setIn(input);
+    }
+
+    @After
+    public void wipeStreams() {
+        System.setOut(systemOutput);
```
Comment (sarahabimay) [inline src/test/java/CliDisplayTest.java:24]: Interested to know why you needed to do this and why the Before isn't enough?

**Pull request merged**